src/Metadata/Resource/Factory/XmlResourceMetadataFactory.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Annotation\Resource;
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use DOMDocument;
+use DOMElement;
+use DOMXpath;
+
+/**
+ * Creates a resource metadata from xml {@see Resource} configuration.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class XmlResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    private $xmlParser;
+    private $paths;
+    private $decorated;
+
+    /**
+     * @param string[]                              $paths
+     * @param ResourceMetadataFactoryInterface|null $decorated
+     */
+    public function __construct(array $paths, ResourceMetadataFactoryInterface $decorated = null)
+    {
+        $this->xmlParser = new DOMDocument();
+        $this->paths = $paths;
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass) : ResourceMetadata
+    {
+        $parentResourceMetadata = null;
+        if ($this->decorated) {
+            try {
+                $parentResourceMetadata = $this->decorated->create($resourceClass);
+            } catch (ResourceClassNotFoundException $resourceNotFoundException) {
+                // Ignore not found exception from decorated factories
+            }
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($resourceClass);
+        } catch (\ReflectionException $reflectionException) {
+            return $this->handleNotFound($parentResourceMetadata, $resourceClass);
+        }
+
+        $metadata = null;
+
+        foreach ($this->paths as $path) {
+            $this->xmlParser->loadXML(file_get_contents($path));
+
+            $xpath = new DOMXpath($this->xmlParser);
```

Comment:
We should create a XML schema to validate the config provided by the user: http://php.net/manual/fr/domdocument.schemavalidate.php